src/main/java/org/thymeleaf/standard/expression/RestrictedJakartaRequestAccessUtils.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * =============================================================================
+ *
+ *   Copyright (c) 2011-2018, The THYMELEAF team (http://www.thymeleaf.org)
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ * =============================================================================
+ */
+package org.thymeleaf.standard.expression;
+
+import org.thymeleaf.exceptions.TemplateProcessingException;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletRequestWrapper;
+import java.util.Map;
+
+/**
+ * <p>
+ *   Utility class that wraps {@link HttpServletRequest} objects in order to restrict access to
+ *   request parameters in specific restricted expression execution environments.
+ * </p>
+ *
+ * @author Daniel Fern&aacute;ndez
+ *
+ * @since 3.0.9
+ *
+ */
+public final class RestrictedJakartaRequestAccessUtils {
```

Comment:
For instance, if the `org.thymeleaf.web.servlet.*` approach was feasible, creating this new version of the restriction tests (basically containing duplicates of the restricted code detection code) would not be needed.

Reply:
This is true. However, I think the disadvantages I listed above outweigh the benefits.